starknet/src/server/stream.rs

```diff
@@ -74,6 +79,31 @@ where
 
         ResponseStream::new(data_stream).instrument(stream_span)
     }
+
+    async fn sync_status(self, rpc_url: Url) -> StatusResponse {
+        let provider = HttpProvider::new(rpc_url);
+        let latest_apibara_block = self.storage.highest_accepted_block().unwrap().unwrap();
```

Comment:
```suggestion
        let node_accepted_block = self.storage.highest_accepted_block()?;
        let node_finalized_block = self.storage.highest_finalized_block()?;
```

It's possible for the accepted block to be `None` (for example, when still syncing old data).  This is also the right place to read the latest finalized block.